### PR TITLE
fix address flag usage

### LINKED ISSUE #88458: net.Listen listens on all available unicast and anycast IP addresses regarding the IP family
**What happened**:

There are flag descriptions where is stated that the unspecified address behaves differently depending on the IP family:

```
(set to 0.0.0.0 for all IPv4 interfaces and :: for all IPv6 interfaces)
```

However, that's not the behavior implemented in `net.Listen`  and `net.ListenPacket`, clarified in the golang documentation https://github.com/golang/go/commit/424b0654f8e6c72f69e096f69009096de16e30fa

> // For TCP networks, if the host in the address parameter is empty or
> // a literal unspecified IP address, Listen listens on all available
> // unicast and anycast IP addresses of the local system.
> // To only use IPv4, use network "tcp4".

**What you expected to happen**:

Fix the comment to state the reality,  i.e
```
 (set 0.0.0.0 or :: for all interfaces")
```

**How to reproduce it (as minimally and precisely as possible)**:

From the root of the kubernetes repo
```sh
grep -r "for all IPv4 interfaces"
...
cmd/kubelet/app/options/options.go:     fs.Var(utilflag.IPVar{Val: &c.HealthzBindAddress}, "healthz-bind-address", "The IP address for the healthz server to serve on (set to `0.0.0.0` for all IPv4 interfaces and `::` for all IPv6 interfaces)")
cmd/kube-proxy/app/server.go:   fs.Var(utilflag.IPVar{Val: &o.config.BindAddress}, "bind-address", "The IP address for the proxy server to serve on (set to `0.0.0.0` for all IPv4 interfaces and `::` for all IPv6 interfaces)")
cmd/kube-proxy/app/server.go:   fs.Var(utilflag.IPVar{Val: &o.config.HealthzBindAddress}, "healthz-bind-address", "The IP address for the health check server to serve on (set to `0.0.0.0` for all IPv4 interfaces and `::` for all IPv6 interfaces)")
```

**Anything else we need to know?**:

xref https://github.com/kubernetes/kubernetes/pull/86830

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by notpad**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
net.Listen listens on all available unicast and anycast IP addresses regarding the IP family,  current description for some address flags are wrong

**Which issue(s) this PR fixes**:

Fixes #88458

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cmd/kube-proxy/app/server.go [hunk 1]
```diff
@@ -165,9 +165,9 @@ func (o *Options) AddFlags(fs *pflag.FlagSet) {
 	fs.BoolVar(&o.CleanupIPVS, "cleanup-ipvs", o.CleanupIPVS, "If true and --cleanup is specified, kube-proxy will also flush IPVS rules, in addition to normal cleanup.")
 	fs.MarkDeprecated("cleanup-ipvs", "In a future release, running --cleanup will always flush IPVS rules")
 
-	fs.Var(utilflag.IPVar{Val: &o.config.BindAddress}, "bind-address", "The IP address for the proxy server to serve on (set to `0.0.0.0` for all IPv4 interfaces and `::` for all IPv6 interfaces)")
-	fs.Var(utilflag.IPVar{Val: &o.config.HealthzBindAddress}, "healthz-bind-address", "The IP address for the health check server to serve on (set to `0.0.0.0` for all IPv4 interfaces and `::` for all IPv6 interfaces)")
-	fs.Var(utilflag.IPVar{Val: &o.config.MetricsBindAddress}, "metrics-bind-address", "The IP address for the metrics server to serve on (set to `0.0.0.0` for all IPv4 interfaces and `::` for all IPv6 interfaces)")
+	fs.Var(utilflag.IPVar{Val: &o.config.BindAddress}, "bind-address", "The IP address for the proxy server to serve on")
```
Comment (aojea) [inline cmd/kube-proxy/app/server.go]: this is explained here
https://github.com/kubernetes/kubernetes/issues/88458#issuecomment-591960548

**Comment by aojea** [comment]
/assign @derekwaynecarr 
for kubelet

**Review by mattjmcnaughton — COMMENTED** [review]
/lgtm

LGTM from a sig-node perspective. To confirm, this diff does not include code updates correct? Its only making the documentation conform to the currently existing code?

**Review by mattjmcnaughton — APPROVED** [review]

**Comment by aojea** [comment]
@notpad I remember reading a better description of the PR (the usage of the `, the kube-proxy bind address, ...) somewhere but I can't find it now, do you mind updating the PR description explaining this changes and answering previous comment?

> To confirm, this diff does not include code updates correct? Its only making the documentation conform to the currently existing code?

**Comment by notpad** [comment]
> /lgtm
> 
> LGTM from a sig-node perspective. To confirm, this diff does not include code updates correct? Its only making the documentation conform to the currently existing code?

Yes, this PR only makes the documentation conform to currently existing code. I think it will be better to open a PR for each change in different sig, will close this PR and open a new one.

**Comment by notpad** [comment]
@mattjmcnaughton a new PR #88731 is opened for kubelet change

**Pull request closed without merging**